plugins/woocommerce-blocks/assets/js/base/components/summary/test/index.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import Summary, { SummaryProps } from '../index';
+
+const allowedSource =
+	'<p class="some-class">' +
+	'Lorem ipsum dolor sit amet ' +
+	'<h1 style="color: red;">Heading</h1>' +
+	'<ul><li>List item</li></ul>' +
+	'<img src="https://example.com/image.jpg" alt="Image" />' +
+	'</p>';
+
+const disallowedSource =
+	'<p class="some-class">' +
+	'Lorem ipsum dolor sit amet ' +
+	'<script src="http://evil.com" />' +
+	'<h1 style="color: red;">Heading</h1>' +
+	'<ul><li>List item</li></ul>' +
+	'<img src="https://example.com/image.jpg" alt="Image" height="100" width="100" />' +
+	'<script>alert("Hello");</script>' +
+	'</p>';
+
+const getProps = ( source: string ) =>
+	( {
+		source,
+		maxLength: 1000,
+		countType: 'words',
+		className: 'test-class',
+	} as SummaryProps );
+
+describe( 'Summary component', () => {
+	it( 'renders rich HTML with the allowed tags and attributes', () => {
+		const props = getProps( allowedSource );
+		const { container } = render( <Summary { ...props } /> );
+
+		expect( container ).toMatchSnapshot( allowedSource );
+	} );
+
+	it( 'omits disallowed tags and attributes', () => {
+		const props = getProps( disallowedSource );
+		const { container } = render( <Summary { ...props } /> );
+
+		expect( container ).toMatchSnapshot( allowedSource );
```

Comment:
@kmanijak  I recommend avoiding snapshot testing. It's hard to keep track of the impact of a change if we rely on snaphots comparison.

Also, you aren't testing for disallowed attributes here, only for tags. It would be good to be explicit about the both cases, and have separate tests, with predictable results that are easy to read from the test.